ocs_ci/ocs/pgsql.py

```diff
@@ -0,0 +1,293 @@
+            if transactions is not None:
+                pg_data[
+                    'spec'
+                ]['workload']['args']['transactions'] = transactions
+            if scaling_factor is not None:
+                pg_data[
+                    'spec'
+                ]['workload']['args']['scaling_factor'] = scaling_factor
+            pg_obj = OCS(**pg_data)
+            pg_obj.create()
+
+        # Confirm that expected pgbench pods are spinned
+        log.info("Checking if Getting pgbench pods name")
+        for pgbench_pods in TimeoutSampler(
+            3600, replicas, get_pod_name_by_pattern,
+            'pgbench-1-dbs-client', RIPSAW_NAMESPACE
+        ):
+            try:
+                if len(pgbench_pods) == replicas:
+                    log.info(
+                        f"Expected number of pgbench pods are "
+                        f"found: {replicas}"
+                    )
+                    break
+            except IndexError:
+                log.info(
+                    f'Expecting number of pgbench pods are {replicas} '
+                    f'but only found {len(pgbench_pods)}'
+                )
+
+    def get_postgresql_pods(self):
+        """
+        Get all postgresql pods
+
+        Returns:
+            List: postgresql pod objects list
+
+        """
+        return get_all_pods(
+            namespace=RIPSAW_NAMESPACE, selector=['postgres']
+        )
+
+    def get_pgbench_pods(self):
+        """
+        Get all pgbench pods
+
+        Returns:
+            List: pgbench pod objects list
+
+        """
+        return [
+            get_pod_obj(
+                pod
+            ) for pod in get_pod_name_by_pattern('pgbench', RIPSAW_NAMESPACE)
+        ]
+
+    def is_pgbench_running(self):
+        """
+        Check if pgbench is running
+
+        Returns:
+            bool: True if pgbench is running; False otherwise
+
+        """
+        pod_objs = self.get_pgbench_pods()
+        for pod in pod_objs:
+            if pod.get().get(
+                'status'
+            ).get('containerStatuses')[0].get('state') == 'running':
+                log.info("One or more pgbench pods are in running state")
+                return True
+            else:
+                return False
+            break
+
+    def get_pgbench_status(self, pgbench_pod_name):
+        """
+        Get pgbench status
+
+        Args:
+            pgbench_pod_name (str): Name of the pgbench pod
+
+        Returns:
+            str: state of pgbench pod (running/completed)
+
+        """
+        pod_obj = get_pod_obj(pgbench_pod_name, namespace=RIPSAW_NAMESPACE)
+        status = pod_obj.get().get(
+            'status'
+        ).get('containerStatuses')[0].get('state')
+        if list(status.keys())[0] == 'running':
+            return 'running'
+        elif list(status.keys())[0] == 'terminated':
+            return status['terminated']['reason']
```

Comment:
Are there any other options for list(status.keys())[0]? if there are, better to have 'else' condition at the end so all conditions will be covered. If not, suggest doing:

```python
        return 'running' if list(status.keys())[0] == 'running' else status['terminated']['reason']
```

Reply:
Done